### PR TITLE
Speedup mdtest parser

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

A few small changes to speed up the mdtest parser:

* Use `memchr` to find possible header and codeblock positions over matching both regexes (fewer LazyLock lookups, skip over more text than line by line)
* Avoid use of `captures` for the header (because captures are "slow" and header parsing is easy)
* Avoid using unicode words for the markdown language

## Test Plan

I "benchmarked" the parser by commenting out the `run_test` invocation. 

Before: ±230ms
Now: ±170ms

### CODE REVIEW

**Review by AlexWaygood — APPROVED** [review]
This all looks fine to me but @carljm is obviously the expert on this code. I haven't benchmarked myself but I trust you when you say it's faster 😄

File: crates/red_knot_test/src/parser.rs
Comment (sharkdp) [inline crates/red_knot_test/src/parser.rs:140]: Minor: the doc-comment should be updated ("Matches an arbitrary amount of whitespace"…)

File: crates/red_knot_test/src/parser.rs
Comment (sharkdp) [inline crates/red_knot_test/src/parser.rs:145]: The new regex does not match empty code blocks such as
````markdown
```py path=a.py
```
````
in `mdtest/import/errors.md`, which is also why some tests are failing. The final newline was optional before: `\n?`.

Comment (AlexWaygood) [reply crates/red_knot_test/src/parser.rs]: It might be worth adding a specific unit test for this edge case in the `red_knot_test` crate, so that we still have tests that will fail even if the mdtest in `mdtest/import/errors.md` changes so it no longer has an empty code block

Comment (MichaReiser) [reply crates/red_knot_test/src/parser.rs]: Agree, I'll follow up, once I have @carljm approval

File: crates/red_knot_test/src/parser.rs
Comment (carljm) [inline crates/red_knot_test/src/parser.rs:247]: I don't think it's worth having a separate `scan` method with only one call site; it's quite short, let's just inline it here. That makes the behavior more parallel to the `HEADER_RE` case (the updating of `self.unparsed` is visible in both branches), which I think makes it easier to understand the logic.

File: crates/red_knot_test/src/parser.rs
Comment (carljm) [inline crates/red_knot_test/src/parser.rs:139]: Do we need the capture groups in this regex if they are not used?

File: crates/red_knot_test/src/parser.rs
Comment (carljm) [inline crates/red_knot_test/src/parser.rs:489]:
```suggestion
            panic!("expected two files");
```

**Review by carljm — APPROVED** [review]
Looks great, thank you!!

File: crates/red_knot_test/src/parser.rs
Comment (MichaReiser) [reply crates/red_knot_test/src/parser.rs:139]: Technically not but there should be no cost because we use `matches`. I do like them for documentation purposes.

**Pull request merged**
